tests/python_package_test/test_engine.py

```diff
@@ -50,6 +50,24 @@ def categorize(continuous_x):
     return np.digitize(continuous_x, bins=np.arange(0, 1, 0.01))
 
 
+@pytest.fixture
+def artifacts_for_refit_kwargs():
```

Comment:
I think that the indirection added by using this fixture isn't worth the small reduction in duplicated code.

Can you please remove this fixture and just copy this code directly into each of the tests you're adding?

Reply:
Actually....as mentioned in https://github.com/microsoft/LightGBM/issues/3038#issuecomment-996211592, please base these new tests on

https://github.com/microsoft/LightGBM/blob/8a34b1af2d5ae734a446f97ee984508a55ecb2d4/tests/python_package_test/test_engine.py#L1531-L1533

Specifically, please use the existing datasets used in other tests.

As I think you've noticed (because of setting `min_data_in_bin=1, min_data_in_leaf=1`), LightGBM has parameter defaults which are not suitable to very very small datasets like this 3-row, 1-feature dataset you've created. Using such small datasets can lead to instability in tests.

Reply:
> Can you please remove this fixture and just copy this code directly into each of the tests you're adding?

I do support this.

Reply:
Also I think these two new tests can be merged into one to reduce overheads.

Reply:
OK, thanks for the reference @jameslamb.